techniques/html/H-creating-dialogs-with-the-html-dialog-element.html

```diff
@@ -0,0 +1,212 @@
+      &lt;h1 id=&quot;dialog-heading&quot; tabindex=&quot;-1&quot;&gt;Sign up to our mailing list&lt;/h1&gt;
+      &lt;form&gt;
+        &lt;p class=&quot;req-note&quot;&gt;All form fields are required.&lt;/p&gt;
+        &lt;div&gt;
+          &lt;label for=&quot;fname&quot;&gt;First Name&lt;/label&gt;
+          &lt;input aria-required=&quot;true&quot; autocomplete=&quot;given-name&quot; id=&quot;fname&quot; type=&quot;text&quot;&gt;
+        &lt;/div&gt;
+        &lt;div&gt;
+          &lt;label for=&quot;lname&quot;&gt;Last Name&lt;/label&gt;
+          &lt;input aria-required=&quot;true&quot; autocomplete=&quot;family-name&quot; id=&quot;lname&quot; type=&quot;text&quot;&gt;
+        &lt;/div&gt;
+        &lt;div&gt;
+          &lt;label for=&quot;email&quot;&gt;Email address&lt;/label&gt;
+          &lt;input aria-required=&quot;true&quot; autocomplete=&quot;email&quot; id=&quot;email&quot; type=&quot;text&quot;&gt;
+        &lt;/div&gt;
+        &lt;button class=&quot;sign-up&quot; type=&quot;submit&quot;&gt;Sign up&lt;/button&gt;
+      &lt;/form&gt;
+      &lt;button aria-label=&quot;close&quot; class=&quot;close-modal&quot; type=&quot;button&quot;&gt;&amp;times;&lt;/button&gt;
+    &lt;/dialog&gt;
+  &lt;/body&gt;
+&lt;/html&gt;</code></pre>
+         
+<h4>The CSS</h4>
+         
+<pre xml:space="preserve"><code class="language-css">*, *::after, *::before {
+  box-sizing: inherit;
+}
+
+body {
+  background:#fff;
+  color:#000;
+  font:1rem/1.5 system-ui, Helvetica, Roboto, sans-serif;
+}
+
+*:focus-visible{
+  outline:1px solid #0054AE;
+  outline-offset:1px;
+}
+
+dialog {
+  border:1px solid #000;
+  padding:2rem;
+  position:relative;
+}
+ 
+dialog::backdrop {
+  background-color:hsla(0, 0%, 0%, .5);
+}
+ 
+.close-modal {
+  inset-block-start:1.5rem;  
+  inset-inline-end:1.5rem;  
+  line-height:1.3;
+  padding:0.25em 0.5em;
+  position:absolute;
+}
+
+.sign-up{
+  background:#000;
+  color:#fff;
+  padding:0.25em;
+}
+
+dialog h1 {
+  display:inline-block;
+  line-height:1.3333;
+  margin:0;
+  max-inline-size:95%;
+}
+
+form{
+  display:grid;
+  grid-gap:20px;
+  grid-template-columns:repeat(auto-fit, minmax(150px, 1fr));
+}
+
+.req-note, .sign-up{
+  grid-column:1 / -1;
+}
+ 
+label {
+  display:block;
+}
+ 
+input {
+  border:1px solid hsl(0, 0%, 50%);
+  font:inherit;
+  inline-size:calc(100% - 4px);
+}
+
+button{
+  background:#fff;
+  border:1px solid hsl(0, 0%, 50%);
+  border-radius:3px;
+  color:inherit;
+  font:inherit;
+  margin:0;
+}</code></pre>
+     
+		<h4>The JavaScript</h4>
+		<p>The script toggles the display of the <code class="language-html">dialog</code>. The HTML <code class="language-html">dialog</code> element can be opened using two different commands: <code class="language-javascript">show()</code> (for non-modal <code class="language-html">dialog</code>s), and  <code class="language-javascript">showModal()</code> (for modal <code class="language-html">dialog</code>s).</p>
```

Comment:
```suggestion
		<p>The script is necessary to display the <code class="language-html">dialog</code> when invoked. The HTML <code class="language-html">dialog</code> element can be opened using two different commands: <code class="language-javascript">show()</code> (for non-modal <code class="language-html">dialog</code>s), and  <code class="language-javascript">showModal()</code> (for modal <code class="language-html">dialog</code>s) - used in this example.</p>
		 <p><strong>Note:</strong> instead of writing a function to close the <code class="language-html">dialog</code>, the close button is nested within a separate <code class="language-html">form</code> element with a <code class="language-html">method="dialog"</code> attribute. This method allows the dialog to be closed when activating this button, and the browser will handle returning keyboard focus to the invoking element, </p>
```